### PR TITLE
fix(textarea): modern template autogrow

### PR DESCRIPTION
**Pull request by sean-perkins**
## Pull request checklist

Please check if your PR fulfills the following requirements:
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been reviewed and added / updated if needed (for bug fixes / features)
  - Some docs updates need to be made in the `ionic-docs` repo, in a separate PR. See the [contributing guide](https://github.com/ionic-team/ionic-framework/blob/main/.github/CONTRIBUTING.md#modifying-documentation) for details.
- [x] Build (`npm run build`) was run locally and any changes were pushed
- [x] Lint (`npm run lint`) has passed locally and any fixes were made for failures

## Pull request type

 

 

Please check the type of change your PR introduces:
- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe): 

## What is the current behavior?

Autogrow is not functional with the modern template rendering of `ion-textarea`. 

Issue URL: N/A

## What is the new behavior?

- Textarea resizing with content changes works with modern template rendering and different fill styles of `ion-textarea`
- iOS cloned input behavior supports new `ion-textarea` autogrow 
- Stacked/floating label adds padding bottom to the container

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

## Other information

- Input spec test diff appears to be a placeholder color diff (unsure why Playwright captured it on this run vs. when that change happened)
- Input items test diff is placeholder color diff

### CODE REVIEW

**Comment by sean-perkins** [comment]
Opening this for review so that we can swarm anything incorrect.

File: core/src/components/textarea/test/autogrow/index.html [hunk 1]
```diff
@@ -0,0 +1,61 @@
+<!DOCTYPE html>
```
Comment (liamdebeasi) [inline core/src/components/textarea/test/autogrow/index.html:1]: Threading:

What are the input legacy tests updated?

File: core/src/components/textarea/test/autogrow/index.html [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+<html lang="en" dir="ltr">
+  <head>
```
Comment (liamdebeasi) [inline core/src/components/textarea/test/autogrow/index.html:3]: A lot of the other tests seem to have visual diffs too. Is this caused the `--padding-bottom: 8px` addition?

File: core/src/components/textarea/textarea.scss [hunk 3]
```diff
@@ -71,8 +73,6 @@
   flex: 1;
 
   background: var(--background);
-
```
Comment (liamdebeasi) [inline core/src/components/textarea/textarea.scss:74]: Why did this move?

File: core/src/components/textarea/textarea.scss [hunk 4]
```diff
@@ -208,7 +187,12 @@
   opacity: 1;
 }
 
-:host([auto-grow="true"]) .cloned-input {
+:host(.legacy-textarea[auto-grow]) .cloned-input {
+  @include margin(0, 0, 0, 0);
+}
+
+:host([auto-grow]) .cloned-input {
+  @include margin(28px, 0, 0, 0);
```
Comment (liamdebeasi) [inline core/src/components/textarea/textarea.scss:195]: What does this do?

File: core/src/components/textarea/textarea.scss [hunk 5]
```diff
@@ -179,6 +147,17 @@
   }
 }
 
+.native-textarea,
+:host(.legacy-textarea) .textarea-legacy-wrapper::after {
+  @include padding(var(--padding-top), var(--padding-end), var(--padding-bottom), var(--padding-start));
```
Comment (liamdebeasi) [inline core/src/components/textarea/textarea.scss:152]: This is already set here https://github.com/ionic-team/ionic-framework/blob/FW-2764/core/src/components/textarea/textarea.scss#L284

Also why do we set `text-inherit()`?

File: core/src/components/textarea/textarea.scss [hunk 6]
```diff
@@ -179,6 +147,17 @@
...
+  @include text-inherit();
+  @include border-radius(var(--border-radius));
```
Comment (liamdebeasi) [inline core/src/components/textarea/textarea.scss:154]: This is already set here: https://github.com/ionic-team/ionic-framework/blob/FW-2764/core/src/components/textarea/textarea.scss#L244

**Comment by sean-perkins** [comment]
It looks like the `--padding-bottom` change causes the floating label alignment to be incorrect. The reason for introducing it was to add padding to the bottom of the textarea container. If you remove it and look at the difference between legacy textarea and modern (e.g. the autogrow templates) you'll see with modern the text runs into the bottom border where as the legacy has spacing at the bottom.

File: core/src/components/textarea/textarea.scss (hunk 3 above)
Comment (sean-perkins) [reply core/src/components/textarea/textarea.scss:74]: This was just clean-up, aligning closer to the original property ordering:
- https://github.com/ionic-team/ionic-framework/blob/feature-7.0/core/src/components/textarea/textarea.scss#L48 (feature-7.0)
- https://github.com/ionic-team/ionic-framework/blob/main/core/src/components/textarea/textarea.scss#L48 (main)

File: core/src/components/textarea/test/autogrow/index.html (hunk 1 above)
Comment (sean-perkins) [reply core/src/components/textarea/test/autogrow/index.html:1]: The placeholder text color changes between the diffs, but unrelated to these changes as far as I am aware.

**Review by liamdebeasi — APPROVED** [review]
Great job!

**Pull request merged**